pkg/generators/duck_type.go

```diff
@@ -0,0 +1,155 @@
+package generators
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/argoproj/argo-cd/util/settings"
+	log "github.com/sirupsen/logrus"
+
+	argoprojiov1alpha1 "github.com/argoproj-labs/applicationset/api/v1alpha1"
+	"github.com/argoproj-labs/applicationset/pkg/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+var _ Generator = (*DuckTypeGenerator)(nil)
+
+// DuckTypeGenerator generates Applications for some or all clusters registered with ArgoCD.
+type DuckTypeGenerator struct {
+	ctx             context.Context
+	dynClient       dynamic.Interface
+	clientset       kubernetes.Interface
+	namespace       string // namespace is the Argo CD namespace
+	settingsManager *settings.SettingsManager
+}
+
+func NewDuckTypeGenerator(ctx context.Context, dynClient dynamic.Interface, clientset kubernetes.Interface, namespace string) Generator {
+
+	settingsManager := settings.NewSettingsManager(ctx, clientset, namespace)
+
+	g := &DuckTypeGenerator{
+		ctx:             ctx,
+		dynClient:       dynClient,
+		clientset:       clientset,
+		namespace:       namespace,
+		settingsManager: settingsManager,
+	}
+	return g
+}
+
+func (g *DuckTypeGenerator) GetRequeueAfter(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator) time.Duration {
+
+	// Return a requeue default of 3 minutes, if no override is specified.
+
+	if appSetGenerator.DuckType.RequeueAfterSeconds != nil {
+		return time.Duration(*appSetGenerator.DuckType.RequeueAfterSeconds) * time.Second
+	}
+
+	return DefaultRequeueAfterSeconds
+}
+
+func (g *DuckTypeGenerator) GetTemplate(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator) *argoprojiov1alpha1.ApplicationSetTemplate {
+	return &appSetGenerator.DuckType.Template
+}
+
+func (g *DuckTypeGenerator) GenerateParams(
+	appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator,
+	_ *argoprojiov1alpha1.ApplicationSet) ([]map[string]string, error) {
```

Comment:
should be in one line
```suggestion
func (g *DuckTypeGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator, _ *argoprojiov1alpha1.ApplicationSet) ([]map[string]string, error) {
```